src/Frontend/test/specs/customchecks/viewing-no-data.spec.ts

```diff
@@ -1,21 +1,47 @@
 import { test, describe } from "../../drivers/vitest/driver";
+import { expect } from "vitest";
+import * as precondition from "../../preconditions";
+import { customChecksMessage } from "./questions/customChecksMessage";
+import { waitFor } from "@testing-library/vue";
 
 describe("FEATURE: No data", () => {
   describe("RULE: When there is no data to show a message should be displayed ", () => {
-    test.todo("EXAMPLE: 'No failed custom checks' should be displayed when there are no custom checks");
-
     /* SCENARIO
           Given there are no custom checks
           When navigating to the custom checks tab
           Then a message is shown "No failed custom checks"
         */
+    test("EXAMPLE: 'No failed custom checks' should be displayed when there are no custom checks", async ({ driver }) => {
+      //Arrange
+      await driver.setUp(precondition.serviceControlWithMonitoring);
+      // given that the custom check list is empty
+      await driver.setUp(precondition.hasCustomChecksEmpty);
 
-    test.todo("EXAMPLE: 'No failed custom checks' should be displayed when all custom checks are in a success state");
+      //Act - When navigating to the custom checks tab
+      await driver.goTo("/custom-checks");
+      //Expect
+      await waitFor(async () => {
+        expect(await customChecksMessage()).toBe("No failed custom checks");
+      });
+    });
     /* SCENARIO
           Given there are custom checks
           And all custom checks are in a success state
           When navigating to the custom checks tab
-          The a message is shown "No failed custom checks"
+          Then a message is shown "No failed custom checks"
         */
+    test("EXAMPLE: 'No failed custom checks' should be displayed when all custom checks are in a success state", async ({ driver }) => {
+      //Arrange
+      await driver.setUp(precondition.serviceControlWithMonitoring);
+      // given that the custom check list is not empty and status of all checks are passing
+      await driver.setUp(precondition.hasCustomChecksPassing);
```

Comment:
To help detect potential regressions in the future, the backend should return a combination of both passing and failing custom checks when the URL being hit does not include status=fail. This approach ensures that the test will fail if:

- A bug is introduced causing the implementation to stop using status=fail.
- The filtering of custom checks is shifted from the server to the browser but the new implementation fails to exclude the passing checks.